snapcraft/models/project.py

```diff
@@ -615,41 +611,40 @@ class Project(models.Project):
 
     # snapcraft's `name` is more general than craft-application
     name: ProjectName  # type: ignore[assignment]
-    build_base: Optional[str]
+    build_base: str | None = pydantic.Field(validate_default=True, default=None)
     compression: Literal["lzo", "xz"] = "xz"
-    version: Optional[VersionStr]  # type: ignore[assignment]
-    donation: Optional[UniqueStrList]
+    version: VersionStr | None = None
+    donation: UniqueStrList | None = None
     # snapcraft's `source_code` is more general than craft-application
-    source_code: Optional[UniqueStrList]  # type: ignore[assignment]
-    contact: Optional[UniqueStrList]  # type: ignore[assignment]
-    issues: Optional[UniqueStrList]  # type: ignore[assignment]
-    website: Optional[UniqueStrList]
-    type: Optional[Literal["app", "base", "gadget", "kernel", "snapd"]]
-    icon: Optional[str]
+    source_code: UniqueStrList | None = None  # type: ignore[assignment]
+    contact: UniqueStrList | None = None  # type: ignore[assignment]
+    issues: UniqueStrList | None = None  # type: ignore[assignment]
+    website: UniqueStrList | None = None
+    type: Literal["app", "base", "gadget", "kernel", "snapd"] | None = None
+    icon: str | None = None
     confinement: Literal["classic", "devmode", "strict"]
-    layout: Optional[
-        Dict[str, Dict[Literal["symlink", "bind", "bind-file", "type"], str]]
-    ]
-    grade: Optional[Literal["stable", "devel"]]
-    architectures: List[Union[str, Architecture]] | None = None
+    layout: (
+        dict[str, dict[Literal["symlink", "bind", "bind-file", "type"], str]] | None
+    ) = None
+    grade: Literal["stable", "devel"] | None = None
```

Comment:
What's the difference between this being this vs. it being:

```suggestion
    grade: Literal["stable", "devel"] = "stable
```

The only thing I can think of is that here we know if `grade` is undefined, but do we use that?

Reply:
Grade gets set on when creating `meta/snap.yaml`: https://github.com/canonical/snapcraft/blob/8ab7fd0c8a1d3f13045bec41a6e0158c063faa9b/snapcraft/meta/snap_yaml.py#L428

I think your change is safe but I'm apprehensive to make it in this PR.